files/synchronize.py

```diff
@@ -173,6 +173,8 @@
    - Expect that dest=~/x will be ~<remote_user>/x even if using sudo.
    - To exclude files and directories from being synchronized, you may add 
      C(.rsync-filter) files to the source directory.
+   - Does not use SSH agent, authentication must be done using keys using a
```

Comment:
I had ssh-agent running, and when I ran ssh, Ansible, or rsync directly on the command-line, all three of them would use it to perform the login.

However, when I used the Ansible synchronize task it would hang, presumably waiting for terminal input. If I added a passphraseless default key, id_rsa, then it was able to successfully login. If I added the PasswordAuthentication=No option (and removed id_rsa key), then it would fail to log in and report an error.

The PasswordAuthentication=no option instructs SSH to not use password authentication, and only use cryptographic key authentication. This is fine because using passwords requires a prompt which caused the hang I observed.

My reason for the note "Does not use SSH agent" was due to the fact that the synchronize module did not discover and use my SSH agent. I hypothesise that it couldn't find the SSH agent because the environment had been modified and connection details weren't available to the subprocess.